dispatch/static/manager/src/js/components/ContentEditor/embeds/VideoEmbed.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+
+import { FormInput, TextInput } from '../../inputs'
+
+class VideoEmbedComponent extends React.Component {
+
+  constructor(props){
+    super(props)
+
+    this.handleUrlChange = this.handleUrlChange.bind(this)
+    this.insertVideo = this.insertVideo.bind(this)
+
+    this.state = {
+      inserted: false,
+      validURL: false
+    }
+  }
+
+  handleUrlChange(e){
+    e.preventDefault()
+    this.props.updateField('url', e.target.value)
+    console.log("URL IS: "+this.props.data.url);
+
+  }
+
+
+  insertVideo(){
+    var id = this.validYouTube(this.props.data.url);
+    this.props.updateField('id',id)
+    console.log("ID IS: "+this.props.data.id)
+
+    if(this.props.data.id){
+      this.setState({
+        validURL: true,
+        inserted : true
+      })
+    }
+  }
+
+  validYouTube(url){
+     var p = /^(?:https?:\/\/)?(?:www\.)?youtube\.com\/watch\?(?=.*v=((\w|-){11}))(?:\S+)?$/
```

Comment:
Ideally it should accept both. I would guess that most people would just copy paste the full URL